cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -74,6 +74,10 @@ type BicepProvider struct {
 	alphaFeatureManager   *alpha.FeatureManager
 	clock                 clock.Clock
 	ignoreDeploymentState bool
+	// compileBicepResult is cached to avoid recompiling the same bicep file multiple times in the same azd run.
```

Comment:
Would it be easy to separate the memory cache changes out?

Reply:
I needed here, :(

I need to prompt for bicep-required-parameters when running `azd pipeline config` so users can save the prompt on evn-config and put it into the pipeline.

Without the cache, if the user rejects the `persist your answer` to env-conf when running `azd up` locally, there would be a second prompt for the same parameter next.
The first prompt, during initialization would be lost.